### ISSUE TITLE
There should be an error when there is an error in emitting .d.ts

### ISSUE THREAD

**Issue by yuit** [Microsoft/TypeScript#7354]
**TypeScript Version:** 

Built from "master" branch

**Code**

```ts
namespace V {
    interface Foo { }
    export function Bar(): Foo { return undefined; }
}
```
Compiler option: ` --declaration`

**Expected behavior:**
There should be an error indicating that there is an error in emitting .d.ts file.

**Actual behavior:** 
There is no error shown. Though it behavour correctly, not emitting .d.ts
